### ISSUE TITLE
Segfault  on exit

### ISSUE THREAD

**Issue by Eisfreak7** [The-Compiler/qutebrowser#1560]
Every time I close qutebrowser it ends with the message `segmentation fault (core dumped)`.  
It doesn't really hurt anything, since everything works normally, but Its obviously not the best way to go out
Here is the stack trace:
```
Process 29336 (qutebrowser) of user 1000 dumped core.

Stack trace of thread 29342:
#0  0x00007f56e483976f _ZN7QObject10disconnectEPKS_PKcS1_S3_ (libQt5Core.so.5)
#1  0x00007f56cf0550f0 n/a (libQt5DBus.so.5)
#2  0x00007f56cf045b72 n/a (libQt5DBus.so.5)
#3  0x00007f56e46321d8 n/a (libQt5Core.so.5)
#4  0x00007f56e928d474 start_thread (libpthread.so.0)
#5  0x00007f56e8fcc69d __clone (libc.so.6)

Stack trace of thread 29340:
#0  0x00007f56e929309f pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
#1  0x00007f56d90f07db n/a (libQt5WebKit.so.5)
#2  0x00007f56d90f0819 n/a (libQt5WebKit.so.5)
#3  0x00007f56e928d474 start_thread (libpthread.so.0)
#4  0x00007f56e8fcc69d __clone (libc.so.6)

Stack trace of thread 29343:
#0  0x00007f56e8fc368d poll (libc.so.6)
#1  0x00007f56e34b2fd6 n/a (libglib-2.0.so.0)
#2  0x00007f56e34b30ec g_main_context_iteration (libglib-2.0.so.0)
#3  0x00007f56e485e72f _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5)
#4  0x00007f56e48064ca _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5)
#5  0x00007f56e462d2f3 _ZN7QThread4execEv (libQt5Core.so.5)
#6  0x00007f56e46321d8 n/a (libQt5Core.so.5)
#7  0x00007f56e928d474 start_thread (libpthread.so.0)
#8  0x00007f56e8fcc69d __clone (libc.so.6)

Stack trace of thread 29344:
#0  0x00007f56e929309f pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
#1  0x00007f56d8defd84 n/a (libQt5WebKit.so.5)
#2  0x00007f56d9120481 n/a (libQt5WebKit.so.5)
#3  0x00007f56e928d474 start_thread (libpthread.so.0)
#4  0x00007f56e8fcc69d __clone (libc.so.6)

Stack trace of thread 29345:
#0  0x00007f56e929309f pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
#1  0x00007f56d8df0df3 n/a (libQt5WebKit.so.5)
#2  0x00007f56d9120481 n/a (libQt5WebKit.so.5)
#3  0x00007f56e928d474 start_thread (libpthread.so.0)
#4  0x00007f56e8fcc69d __clone (libc.so.6)

Stack trace of thread 29346:
#0  0x00007f56e929309f pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
#1  0x00007f56d8df0df3 n/a (libQt5WebKit.so.5)
#2  0x00007f56d9120481 n/a (libQt5WebKit.so.5)
#3  0x00007f56e928d474 start_thread (libpthread.so.0)
#4  0x00007f56e8fcc69d __clone (libc.so.6)

Stack trace of thread 29347:
#0  0x00007f56e929309f pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
#1  0x00007f56d8df0df3 n/a (libQt5WebKit.so.5)
#2  0x00007f56d9120481 n/a (libQt5WebKit.so.5)
#3  0x00007f56e928d474 start_thread (libpthread.so.0)
#4  0x00007f56e8fcc69d __clone (libc.so.6)

Stack trace of thread 29338:
#0  0x00007f56e8fc368d poll (libc.so.6)
#1  0x00007f56ddce98e0 n/a (libxcb.so.1)
#2  0x00007f56ddceb679 xcb_wait_for_event (libxcb.so.1)
#3  0x00007f56d0be4bc9 n/a (libQt5XcbQpa.so.5)
#4  0x00007f56e46321d8 n/a (libQt5Core.so.5)
#5  0x00007f56e928d474 start_thread (libpthread.so.0)
#6  0x00007f56e8fcc69d __clone (libc.so.6)

Stack trace of thread 29336:
#0  0x00007f56e929309f pthread_cond_wait@@GLIBC_2.3.2 (libpthread.so.0)
#1  0x00007f56e463308b _ZN14QWaitCondition4waitEP6QMutexm (libQt5Core.so.5)
#2  0x00007f56e4631d8e _ZN7QThread4waitEm (libQt5Core.so.5)
#3  0x00007f56cf0458d6 n/a (libQt5DBus.so.5)
#4  0x00007f56cf045969 n/a (libQt5DBus.so.5)
#5  0x00007f56e8f1abe8 __run_exit_handlers (libc.so.6)
#6  0x00007f56e8f1ac35 exit (libc.so.6)
#7  0x00007f56e95e917f Py_Exit (libpython3.5m.so.1.0)
#8  0x00007f56e95ebe68 n/a (libpython3.5m.so.1.0)
#9  0x00007f56e95ec23d PyErr_PrintEx (libpython3.5m.so.1.0)
#10 0x00007f56e95ed635 PyRun_SimpleFileExFlags (libpython3.5m.so.1.0)
#11 0x00007f56e96040f4 Py_Main (libpython3.5m.so.1.0)
#12 0x0000000000400ae7 main (python3.5)
#13 0x00007f56e8f05741 __libc_start_main (libc.so.6)
#14 0x0000000000400b89 _start (python3.5)
```

Also, unrelated (I'm just mentioning it here instead of its own issue since I can't reliably reproduce it and it may have something to do with my setup instead of qutebrowser, but maybe you know the cause): 
Sometimes when I start qutebrowser it starts in fullscreen mode and I can't exit out of it. The only options are close qutebrowser or move to another desktop. When I open a new qutebrowser window (additional to the existing one in fullscreen) the new window doesn't have the same problem.

**Comment by The-Compiler** [comment]
That's a bug in Qt 5.6, see #1387.

FWIW the [upstream bug](https://bugreports.qt.io/browse/QTBUG-52988) was suddenly closed as fixed in 5.6.1 today, but I haven't had a chance to test yet.

> Also, unrelated (I'm just mentioning it here instead of its own issue since I can't reliably reproduce it and it may have something to do with my setup instead of qutebrowser, but maybe you know the cause): 
> Sometimes when I start qutebrowser it starts in fullscreen mode and I can't exit out of it. The only options are close qutebrowser or move to another desktop. When I open a new qutebrowser window (additional to the existing one in fullscreen) the new window doesn't have the same problem.

I've never seen that or heard anything similar, sorry - might be an issue with your window manager and Qt somehow?

**Comment by Eisfreak7** [comment]
Ah sorry I searched the closed issues but I must've missed that one.

Yes that'll probably be the issue. I'm using bspwm, so that may be the issue. Maybe I'll find a way to track the issue down.

Thank you for your fast reply
